diff --git a/CMakeLists.txt b/CMakeLists.txt
index 87d309c2..338604b7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -252,7 +252,7 @@ else()
 	add_library(Usrsctp::Usrsctp ALIAS usrsctp)
 endif()
 
-add_library(datachannel SHARED
+add_library(datachannel SHARED EXCLUDE_FROM_ALL
 	${LIBDATACHANNEL_SOURCES}
 	${LIBDATACHANNEL_HEADERS}
 	${LIBDATACHANNEL_IMPL_SOURCES}
@@ -271,7 +271,7 @@ endif()
 
 target_compile_definitions(datachannel PRIVATE RTC_EXPORTS)
 
-add_library(datachannel-static STATIC EXCLUDE_FROM_ALL
+add_library(datachannel-static STATIC
 	${LIBDATACHANNEL_SOURCES}
 	${LIBDATACHANNEL_HEADERS}
 	${LIBDATACHANNEL_IMPL_SOURCES}
@@ -439,7 +439,9 @@ if(WARNINGS_AS_ERRORS)
 	endif()
 endif()
 
-install(TARGETS datachannel EXPORT LibDataChannelTargets
+set_target_properties(datachannel-static PROPERTIES OUTPUT_NAME datachannel)
+set_target_properties(juice-static PROPERTIES OUTPUT_NAME juice)
+install(TARGETS datachannel-static juice-static usrsctp plog EXPORT LibDataChannelTargets
 	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
@@ -449,29 +451,6 @@ install(FILES ${LIBDATACHANNEL_HEADERS}
 	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/rtc
 )
 
-# Export targets
-install(
-	EXPORT LibDataChannelTargets
-	FILE LibDataChannelTargets.cmake
-	NAMESPACE LibDataChannel::
-	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/LibDataChannel
-)
-
-# Export config
-install(
-	FILES ${CMAKE_CURRENT_SOURCE_DIR}/cmake/LibDataChannelConfig.cmake
-	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/LibDataChannel
-)
-
-# Export config version
-include(CMakePackageConfigHelpers)
-write_basic_package_version_file(
-	${CMAKE_BINARY_DIR}/LibDataChannelConfigVersion.cmake
-	VERSION ${PROJECT_VERSION}
-	COMPATIBILITY SameMajorVersion)
-install(FILES ${CMAKE_BINARY_DIR}/LibDataChannelConfigVersion.cmake
-	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/LibDataChannel)
-
 # Tests
 if(NOT NO_TESTS)
 	if(CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
