diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4fee473..3e7f5ec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -269,8 +269,7 @@ else()
 endif()
 
 if(NO_MEDIA)
-	target_compile_definitions(datachannel PUBLIC RTC_ENABLE_MEDIA=0)
-	target_compile_definitions(datachannel-static PUBLIC RTC_ENABLE_MEDIA=0)
+	add_compile_definitions(RTC_ENABLE_MEDIA=0)
 else()
 	target_compile_definitions(datachannel PUBLIC RTC_ENABLE_MEDIA=1)
 	target_compile_definitions(datachannel-static PUBLIC RTC_ENABLE_MEDIA=1)
